### PR TITLE
Inconsistent DebuggerDisplay formatting - Fix space, equal for name/value and Indirect Value

### PR DESCRIPTION
**Pull request by whosenbocus**
Addresses https://github.com/dotnet/runtime/issues/89283

### CODE REVIEW

**Comment by whosenbocus** [comment]
@dotnet-policy-service agree

File: src/libraries/System.Speech/src/Result/SemanticValue.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@
 namespace System.Speech.Recognition
 {
     [Serializable]
-    [DebuggerDisplay("'{_keyName}'= {Value}  -  Children = {_dictionary.Count}")]
+    [DebuggerDisplay("'{_keyName}' = {Value}  -  Children = {_dictionary.Count}")]
```
Comment (JamesNK) [inline src/libraries/System.Speech/src/Result/SemanticValue.cs:12]:
```suggestion
    [DebuggerDisplay("{_keyName} = {Value}, Children = {_dictionary.Count}")]
```
`_keyName` is a string so it should already have double quotes.

File: src/libraries/System.Threading/src/System/Threading/Barrier.cs [hunk 2]
```diff
@@ -91,7 +91,7 @@ protected BarrierPostPhaseException(SerializationInfo info, StreamingContext con
     /// completed.
     /// </para>
     /// </remarks>
-    [DebuggerDisplay("Participant Count={ParticipantCount},Participants Remaining={ParticipantsRemaining}")]
+    [DebuggerDisplay("Participant Count = {ParticipantCount}, Participants Remaining = {ParticipantsRemaining}")]
```
Comment (JamesNK) [inline src/libraries/System.Threading/src/System/Threading/Barrier.cs:94]:
```suggestion
    [DebuggerDisplay("ParticipantCount = {ParticipantCount}, ParticipantsRemaining = {ParticipantsRemaining}")]
```
Make names in debugger display match property names.

File: src/libraries/System.Threading/src/System/Threading/CountdownEvent.cs [hunk 3]
```diff
@@ -22,7 +22,7 @@ namespace System.Threading
     /// completed, and Reset, which should only be used when no other threads are
     /// accessing the event.
     /// </remarks>
-    [DebuggerDisplay("Initial Count={InitialCount}, Current Count={CurrentCount}")]
+    [DebuggerDisplay("Initial Count = {InitialCount}, Current Count = {CurrentCount}")]
```
Comment (JamesNK) [inline src/libraries/System.Threading/src/System/Threading/CountdownEvent.cs:25]:
```suggestion
    [DebuggerDisplay("InitialCount = {InitialCount}, CurrentCount = {CurrentCount}")]
```

**Comment by whosenbocus** [comment]
Committed further changes related to indirect value assignment; either as part of variable or method.

**Comment by whosenbocus** [comment]
@danmoseley, @JamesNK 

Let me know your thought on the PR.

Also, do we need to edit the below?


https://github.com/dotnet/runtime/blob/main/src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/FactoryId.cs

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/FactoryRecord.cs

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/LocationSectionRecord.cs

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/SectionInput.cs

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/SectionRecord.cs

**Comment by JamesNK** [comment]
@ericstj @jeffhandley Please assign someone to review this.

**Review by jeffhandley — APPROVED** [review]
The changes look good to me; thanks for the contribution, @whosenbocus.

Before merging this in for .NET 8 RC1 though, I need to confirm there's nothing upstack this could disrupt in tests. Anything come to mind, @stephentoub or @jeffschwMSFT?

**Comment by ericstj** [comment]
I wonder if a test could be added to ensure this formatting is preserved, or perhaps an analyzer?

**Comment by tommcdon** [comment]
+ @gregg-miskelly @caslan

**Review by gregg-miskelly — APPROVED** [review]

**Review by isadorasophia — APPROVED** [review]

**Comment by jeffhandley** [comment]
Thanks for the reviews @isadorasophia and @gregg-miskelly! Without any concerns being raised, I'll go ahead and merge this in for it to be included in .NET 8 RC1.

**Comment by TonyValenti** [comment]
I love changes like this!

I have a few other thoughts, I would like to share regarding debugger displays based off of my own code. What is the best way for me to communicate and share this information?

**Comment by jeffhandley** [comment]
Thanks, @TonyValenti. The best way to share your thoughts on other ways to improve debugger displays is to file a new issue (the [Blank Issue](https://github.com/dotnet/runtime/issues/new) template should be fine), and capture the scenarios you'd like to see improved. An ideal submission would include examples of the experience now and the experience you suggest, citing how the change would benefit you. If there are specific APIs you have in mind, call them out. And if you have generalized input, that's welcomed too.

**Pull request merged**